.github/workflows/ci.yml

```diff
@@ -9,208 +9,248 @@ on:
       - "swagger/**"
 
 env:
-  # TODO: Change variable to your image's name.
   IMAGE_NAME: workbench-server
   DOCKER_BUILDKIT: 1
-
+  REGISTRY: ghcr.io
 
 jobs:
-  # blocked by https://github.com/docker/build-push-action/issues/493
-  # # Build the image and cache it for later use
-  # build: 
-  #   runs-on: ubuntu-latest
-
-  #   steps:
-  #     - name: Checkout
-  #       uses: actions/checkout@v2
-  #       with:
-  #         lfs: true
-
-  #     - name: Ensure buildkit is used
-  #       uses: docker/setup-buildx-action@v1
-  #       id: buildx
-  #       with:
-  #         install: true
-      
-  #     - name: Build and cache
-  #       uses: docker/build-push-action@v2
-  #       with:
-  #         context: .
-  #         push: false
-  #         cache-from: type=gha
-  #         cache-to: type=gha,mode=max
-      
-
-  # Run tests.
-  # See also https://docs.docker.com/docker-hub/builds/automated-testing/
-  test:
+  # Generate test matrix dynamically by dividing spec files into buckets
+  generate-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Cached LFS checkout
+        uses: nschloe/action-cached-lfs-checkout@v1.2.3
+        
+      - name: Generate test matrix
+        id: set-matrix
+        run: |
+          ruby generate-test-matrix.rb > matrix.json
+          echo "matrix=$(cat matrix.json)" >> $GITHUB_OUTPUT
```

Comment:
no need to save this to a file, 

just run it, tee it for debugging, then pipe directly into github output